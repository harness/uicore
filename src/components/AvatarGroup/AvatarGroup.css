.stack {
  display: inline-flex;
  flex-direction: row-reverse;
}
.noStack {
  display: inline-flex;
}
.stackAvatar:not(:last-child) {
  margin-left: var(--avatar-over-lap);
}
