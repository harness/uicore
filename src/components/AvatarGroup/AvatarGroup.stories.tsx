import React from 'react'
import type { Meta, Story } from '@storybook/react'

import { Layout, AvatarGroup, AvatarGroupProps, Color } from '../..'

export default {
  title: 'Components / AvatarGroup',
  component: AvatarGroup
} as Meta
export const Normal: Story<AvatarGroupProps> = args => {
  let { avatars } = args
  if (!avatars) {
    avatars = [
      { name: 'Aahn Boe SS' },
      { email: 'CohnDoe.com.HD@harnes.io' },
      { name: 'EahnFoe' },
      { name: 'Gahn Hoe', backgroundColor: Color.RED_800 },
      { name: '' },
      {
        src: 'https://image.shutterstock.com/image-photo/buddha-statue-hyderabadhyderabadindia-260nw-1366155431.jpg'
      }
    ]
  }
  return (
    <Layout.Horizontal spacing="small" id="primary-buttons">
      <AvatarGroup
        onAdd={() => {
          // eslint-disable-next-line
          alert('add button clicked')
        }}
        {...args}
        size="normal"
        avatars={avatars}
      />
    </Layout.Horizontal>
  )
}
export const Xsmall: Story<AvatarGroupProps> = args => {
  let { avatars } = args
  if (!avatars) {
    avatars = [
      { name: 'Aahn Boe SS' },
      { email: 'CohnDoe.com.HD@harnes.io' },
      { name: 'EahnFoe' },
      { name: 'Gahn Hoe', backgroundColor: Color.RED_800 },
      { name: '' },
      {
        src: 'https://image.shutterstock.com/image-photo/buddha-statue-hyderabadhyderabadindia-260nw-1366155431.jpg'
      }
    ]
  }
  return (
    <Layout.Horizontal spacing="small" id="primary-buttons">
      <AvatarGroup
        onAdd={() => {
          // eslint-disable-next-line
          alert('add button clicked')
        }}
        {...args}
        size="xsmall"
        avatars={avatars}
      />
    </Layout.Horizontal>
  )
}
export const Small: Story<AvatarGroupProps> = args => {
  let { avatars } = args
  if (!avatars) {
    avatars = [
      { name: 'Aahn Boe SS' },
      { email: 'CohnDoe.com.HD@harnes.io' },
      { name: 'EahnFoe' },
      { name: 'Gahn Hoe', backgroundColor: Color.RED_800 },
      { name: '' },
      {
        src: 'https://image.shutterstock.com/image-photo/buddha-statue-hyderabadhyderabadindia-260nw-1366155431.jpg'
      }
    ]
  }
  return (
    <Layout.Horizontal spacing="small" id="primary-buttons">
      <AvatarGroup
        onAdd={() => {
          // eslint-disable-next-line
          alert('add button clicked')
        }}
        {...args}
        size="small"
        avatars={avatars}
      />
    </Layout.Horizontal>
  )
}
export const Large: Story<AvatarGroupProps> = args => {
  let { avatars } = args
  if (!avatars) {
    avatars = [
      { name: 'Aahn Boe SS' },
      { email: 'CohnDoe.com.HD@harnes.io' },
      { name: 'EahnFoe' },
      { name: 'Gahn Hoe', backgroundColor: Color.RED_800 },
      { name: '' },
      {
        src: 'https://image.shutterstock.com/image-photo/buddha-statue-hyderabadhyderabadindia-260nw-1366155431.jpg'
      }
    ]
  }
  return (
    <Layout.Horizontal spacing="small" id="primary-buttons">
      <AvatarGroup
        onAdd={() => {
          // eslint-disable-next-line
          alert('add button clicked')
        }}
        {...args}
        size="large"
        avatars={avatars}
      />
    </Layout.Horizontal>
  )
}
export const Medium: Story<AvatarGroupProps> = args => {
  let { avatars } = args
  if (!avatars) {
    avatars = [
      { name: 'Aahn Boe SS' },
      { email: 'CohnDoe.com.HD@harnes.io' },
      { name: 'EahnFoe' },
      { name: 'Gahn Hoe', backgroundColor: Color.RED_800 },
      { name: '' },
      {
        src: 'https://image.shutterstock.com/image-photo/buddha-statue-hyderabadhyderabadindia-260nw-1366155431.jpg'
      }
    ]
  }
  return (
    <Layout.Horizontal spacing="small" id="primary-buttons">
      <AvatarGroup
        onAdd={() => {
          // eslint-disable-next-line
          alert('add button clicked')
        }}
        {...args}
        size="medium"
        avatars={avatars}
      />
    </Layout.Horizontal>
  )
}
