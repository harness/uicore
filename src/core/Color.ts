export const Color = {
  PRIMARY_BG: 'primaryBG',
  PRIMARY_10: 'primary10',
  PRIMARY_9: 'primary9',
  PRIMARY_8: 'primary8',
  PRIMARY_7: 'primary7',
  PRIMARY_6: 'primary6',
  PRIMARY_5: 'primary5',
  PRIMARY_4: 'primary4',
  PRIMARY_3: 'primary3',
  PRIMARY_2: 'primary2',
  PRIMARY_1: 'primary1',
  GREY_1000: 'grey1000',
  GREY_900: 'grey900',
  GREY_800: 'grey800',
  GREY_700: 'grey700',
  GREY_600: 'grey600',
  GREY_500: 'grey500',
  GREY_400: 'grey400',
  GREY_300: 'grey300',
  GREY_200: 'grey200',
  GREY_100: 'grey100',
  GREY_50: 'grey50',
  GREY_0: 'grey0',
  GREEN_900: 'green900',
  GREEN_800: 'green800',
  GREEN_700: 'green700',
  GREEN_600: 'green600',
  GREEN_500: 'green500',
  GREEN_400: 'green400',
  GREEN_300: 'green300',
  GREEN_200: 'green200',
  GREEN_100: 'green100',
  GREEN_50: 'green50',
  BLUE_900: 'blue900',
  BLUE_800: 'blue800',
  BLUE_700: 'blue700',
  BLUE_600: 'blue600',
  BLUE_500: 'blue500',
  BLUE_400: 'blue400',
  BLUE_300: 'blue300',
  BLUE_200: 'blue200',
  BLUE_100: 'blue100',
  BLUE_50: 'blue50',
  YELLOW_900: 'yellow900',
  YELLOW_800: 'yellow800',
  YELLOW_700: 'yellow700',
  YELLOW_600: 'yellow600',
  YELLOW_500: 'yellow500',
  YELLOW_400: 'yellow400',
  YELLOW_300: 'yellow300',
  YELLOW_200: 'yellow200',
  YELLOW_100: 'yellow100',
  YELLOW_50: 'yellow50',
  ORANGE_900: 'orange900',
  ORANGE_800: 'orange800',
  ORANGE_700: 'orange700',
  ORANGE_600: 'orange600',
  ORANGE_500: 'orange500',
  ORANGE_400: 'orange400',
  ORANGE_300: 'orange300',
  ORANGE_200: 'orange200',
  ORANGE_100: 'orange100',
  ORANGE_50: 'orange50',
  RED_900: 'red900',
  RED_800: 'red800',
  RED_700: 'red700',
  RED_600: 'red600',
  RED_500: 'red500',
  RED_400: 'red400',
  RED_300: 'red300',
  RED_200: 'red200',
  RED_100: 'red100',
  RED_50: 'red50',
  TEAL_900: 'teal900',
  TEAL_800: 'teal800',
  TEAL_700: 'teal700',
  TEAL_600: 'teal600',
  TEAL_500: 'teal500',
  TEAL_400: 'teal400',
  TEAL_300: 'teal300',
  TEAL_200: 'teal200',
  TEAL_100: 'teal100',
  TEAL_50: 'teal50',
  LIME_900: 'lime900',
  LIME_800: 'lime800',
  LIME_700: 'lime700',
  LIME_600: 'lime600',
  LIME_500: 'lime500',
  LIME_400: 'lime400',
  LIME_300: 'lime300',
  LIME_200: 'lime200',
  LIME_100: 'lime100',
  LIME_50: 'lime50',
  PURPLE_900: 'purple900',
  PURPLE_800: 'purple800',
  PURPLE_700: 'purple700',
  PURPLE_600: 'purple600',
  PURPLE_500: 'purple500',
  PURPLE_400: 'purple400',
  PURPLE_300: 'purple300',
  PURPLE_200: 'purple200',
  PURPLE_100: 'purple100',
  PURPLE_50: 'purple50',
  MAGENTA_900: 'magenta900',
  MAGENTA_800: 'magenta800',
  MAGENTA_700: 'magenta700',
  MAGENTA_600: 'magenta600',
  MAGENTA_500: 'magenta500',
  MAGENTA_400: 'magenta400',
  MAGENTA_300: 'magenta300',
  MAGENTA_200: 'magenta200',
  MAGENTA_100: 'magenta100',
  MAGENTA_50: 'magenta50',
  AQUA_500: 'aqua500',
  BLACK: 'black',
  BLACK_100: 'black100',
  WHITE: 'white',
  // Form styles
  FORM_BG: 'form',
  FORM_SECTION_BG: 'formSection',
  FORM_PANEL_BG: 'formPanel',
  FORM_SUBSECTION_BG: 'formSubSection',
  ERROR: 'error',
  WARNING: 'warning',
  SUCCESS: 'success',
  // Extra
  GREY_450: 'grey450',
  GREY_350: 'grey350',
  GREY_250: 'grey250',
  BLUE_450: 'blue450',
  BLUE_350: 'blue350',
  RED_450: 'red450',
  RED_350: 'red350',
  YELLOW_450: 'yellow450',
  YELLOW_350: 'yellow350',
  GREEN_450: 'green450',
  GREEN_350: 'green350',
  SEA_GREEN_500: 'seaGreen500'
}

export type Color = typeof Color[keyof typeof Color]
