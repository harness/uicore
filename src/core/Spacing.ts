// enum Spacing {
//   NONE = 'none',
//   XSMALL = 'xsmall',
//   SMALL = 'small',
//   MEDIUM= 'medium',
//   LARGE= 'large' ,
//   XLARGE= 'xlarge' ,
//   XXLARGE= 'xxlarge' ,
//   XXXLARGE= 'xxxlarge' ,
//   HUGE= 'huge'
// }

export type Spacing = 'none' | 'xsmall' | 'small' | 'medium' | 'large' | 'xlarge' | 'xxlarge' | 'xxxlarge' | 'huge'
