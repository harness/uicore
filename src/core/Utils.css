.tooltipContainer {
  padding: var(--spacing-medium);
  max-width: '500px';
  max-height: '500px';
  overflow: 'auto';
  overflow-wrap: 'break-word';
  word-wrap: 'break-word';
  hyphens: 'auto';
}
