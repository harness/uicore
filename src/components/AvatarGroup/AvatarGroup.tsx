import React, { CSSProperties } from 'react'
import { HTMLDivProps } from '@blueprintjs/core'
import { AvatarProps, Avatar, AvatarSizes } from '../Avatar/Avatar'
import css from './AvatarGroup.css'
import { Color } from '../../core/Color'
import classnames from 'classnames'

export interface AvatarGroupProps extends HTMLDivProps {
  onAdd?: (e: React.MouseEvent<HTMLDivElement>) => void
  avatars: Array<Omit<AvatarProps, 'size'>>
  overlap?: boolean
  avatarGroupProps?: Omit<AvatarProps, 'size'>
  size?: AvatarSizes
  className?: string
}
type overLap = {
  [key in AvatarSizes]: string
}
const avatarOverLap: overLap = { xsmall: '-8px', small: '-10px', large: '-20px', medium: '-13px', normal: '-13px' }
export const AvatarGroup: React.FC<AvatarGroupProps> = ({
  onAdd,
  avatars,
  overlap = true,
  avatarGroupProps,
  size = 'normal',
  className = '',
  ...rest
}) => {
  if (!avatars || !avatars.length) {
    return null
  } else {
    // to avoid mutation of props
    avatars = [...avatars]
  }
  if (onAdd) {
    avatars.push({
      name: '+',
      color: Color.BLUE_500,
      backgroundColor: Color.GREY_200,
      onClick: onAdd,
      hoverCard: false
    })
  }
  if (overlap && avatars.length) {
    /* in order to create a stack like effect where first avatar overlaps second and second the third one and so on,
    we used flex-direction: row-reverse; but this makes the avatars order from right to left and first avatar will
    be the last one to display from left to right, to correct this we are reversing the array elements
    */
    avatars = avatars.reverse()
  }

  const cssProperties = { '--avatar-over-lap': avatarOverLap[size] }
  return (
    <div
      className={classnames(overlap ? css.stack : css.noStack, className)}
      {...rest}
      style={cssProperties as CSSProperties}>
      {avatars.map(avatarProps => (
        <Avatar
          key={Object.values(avatarProps).join('')}
          className={classnames(overlap && css.stackAvatar)}
          {...avatarGroupProps}
          {...avatarProps}
          {...{ size }}
        />
      ))}
    </div>
  )
}
