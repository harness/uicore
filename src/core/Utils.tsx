/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { Classes } from '@blueprintjs/core'
import copy from 'clipboard-copy'
import cx from 'classnames'
import React, { KeyboardEvent, MouseEvent } from 'react'
import { Popover } from '../components/Popover/Popover'
import { Color, Intent, OptionalTooltip } from '@harness/design-system'
import { FormikContext } from 'formik'
import { get, isPlainObject } from 'lodash-es'
import css from './Utils.css'

function stopEvent(event: MouseEvent | KeyboardEvent): void {
  event.stopPropagation()
  event.preventDefault()
}

const randomId = (): string => Math.random().toString(36).substring(2)

function getIntentColors(intent: Intent): { color: string; backgroundColor: string } {
  let color: Color = Color.WHITE
  let backgroundColor

  switch (intent) {
    case Intent.PRIMARY:
      backgroundColor = Color.PRIMARY_7
      break
    case Intent.SUCCESS:
      backgroundColor = Color.GREEN_500
      break
    case Intent.WARNING:
      backgroundColor = Color.YELLOW_500
      break
    case Intent.DANGER:
      backgroundColor = Color.RED_500
      break
    default:
      color = Color.GREY_500
      backgroundColor = Color.WHITE
      break
  }

  return { color, backgroundColor }
}

// Convert UICore named color into real CSS color
const getRealCSSColor = (color: Color): string =>
  `var(--${color // eslint-disable-line
    .match(/[A-Z][a-z]+|[0-9]+|[a-z]+/g)!
    .join('-')
    .toLowerCase()})`

interface WrapOptionalTooltipProps extends OptionalTooltip {
  children: JSX.Element
}

export function WrapOptionalTooltip({ tooltip, tooltipProps, children }: WrapOptionalTooltipProps): React.ReactElement {
  const isDark = tooltipProps?.isDark
  const content =
    typeof tooltip === 'string' ? (
      <div className={css.tooltipContainer} color={(isDark && 'white') || undefined}>
        {tooltip}
      </div>
    ) : (
      tooltip
    )

  return tooltip ? (
    <Popover
      boundary="viewport"
      position="top"
      interactionKind="hover"
      {...tooltipProps}
      popoverClassName={cx(isDark ? Classes.DARK : undefined, tooltipProps?.popoverClassName)}
      content={content || ''}>
      {children}
    </Popover>
  ) : (
    children
  )
}

export const errorCheck = (name: string, formik?: FormikContext<any>): boolean | '' | 0 | undefined =>
  ((get(formik?.touched, name) || (formik?.submitCount && formik?.submitCount > 0)) &&
    get(formik?.errors, name) &&
    !isPlainObject(get(formik?.errors, name))) as boolean

export function romanize(num: number, isLowerCase = false): string | number {
  if (isNaN(num)) return NaN
  const digits = String(+num).split('')
  const key = [
    '',
    'C',
    'CC',
    'CCC',
    'CD',
    'D',
    'DC',
    'DCC',
    'DCCC',
    'CM',
    '',
    'X',
    'XX',
    'XXX',
    'XL',
    'L',
    'LX',
    'LXX',
    'LXXX',
    'XC',
    '',
    'I',
    'II',
    'III',
    'IV',
    'V',
    'VI',
    'VII',
    'VIII',
    'IX'
  ]

  let roman = ''
  let i = 3
  while (i--) roman = (key[parseInt(digits.pop() || '') + i * 10] || '') + roman
  const returnStr = Array(+digits.join('') + 1).join('M') + roman
  return isLowerCase ? returnStr.toLowerCase() : returnStr
}

export const getSelectComponentPlaceholder = (plc?: string) => {
  // idea is to display placeholder like `- Select -`
  // This is as per the latest designs
  if (plc) {
    return `- ${plc} -`
  }
  return `- Select -`
}

// adopted from https://github.com/sindresorhus/escape-string-regexp v5.0.0
export function escapeStringRegexp(str: string): string {
  // Escape characters with special meaning either inside or outside character sets.
  // Use a simple backslash escape when it’s always valid, and a `\xnn` escape when the simpler form would be disallowed by Unicode patterns’ stricter grammar.
  return str.replace(/[|\\{}()[\]^$+*?.]/g, '\\$&').replace(/-/g, '\\x2d')
}

const Utils = {
  stopEvent,
  copy,
  randomId,
  getIntentColors,
  getRealCSSColor,
  WrapOptionalTooltip,
  romanize,
  getSelectComponentPlaceholder,
  escapeStringRegexp
}

export { Utils }
