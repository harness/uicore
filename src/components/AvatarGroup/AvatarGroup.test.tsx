import React from 'react'
import { render } from '@testing-library/react'
import { AvatarGroup } from './AvatarGroup'
import { Color } from '../../core/Color'

describe('Render basic component', () => {
  test('should check snapshot with Avatar props with add button', () => {
    const { container } = render(
      <AvatarGroup
        onAdd={() => {}}
        avatars={[
          { name: 'Aahn Boe SS', backgroundColor: Color.RED_100 },
          { email: 'CohnDoe.com.HD@harnes.io', backgroundColor: Color.RED_100 },
          { name: 'Eahn Foe', backgroundColor: Color.RED_100 },
          { name: 'Gahn Hoe', backgroundColor: Color.RED_100 },
          {
            src:
              'https://image.shutterstock.com/image-photo/buddha-statue-hyderabadhyderabadindia-260nw-1366155431.jpg',
            backgroundColor: Color.RED_100
          }
        ]}
      />
    )
    expect(container).toMatchSnapshot()
  })
  test('should check snapshot with Avatar props without add button', () => {
    const { container } = render(
      <AvatarGroup
        avatars={[
          { name: 'Aahn Boe SS', backgroundColor: Color.RED_100 },
          { email: 'CohnDoe.com.HD@harnes.io', backgroundColor: Color.RED_100 },
          { name: 'EahnFoe', backgroundColor: Color.RED_100 },
          { name: 'Gahn Hoe', backgroundColor: Color.RED_100 },
          { name: '' },
          {
            src:
              'https://image.shutterstock.com/image-photo/buddha-statue-hyderabadhyderabadindia-260nw-1366155431.jpg',
            backgroundColor: Color.RED_100
          }
        ]}
      />
    )
    expect(container).toMatchSnapshot()
  })
  test('should check snapshot with Avatar props without overlap', () => {
    const { container } = render(
      <AvatarGroup
        overlap={false}
        avatars={[
          { name: 'Aahn Boe SS', backgroundColor: Color.RED_100 },
          { email: 'CohnDoe.com.HD@harnes.io', backgroundColor: Color.RED_100 },
          { name: 'EahnFoe', backgroundColor: Color.RED_100 },
          { name: 'Gahn Hoe', backgroundColor: Color.RED_100 },
          { name: '' },
          {
            src:
              'https://image.shutterstock.com/image-photo/buddha-statue-hyderabadhyderabadindia-260nw-1366155431.jpg',
            backgroundColor: Color.RED_100
          }
        ]}
      />
    )
    expect(container).toMatchSnapshot()
  })
})
