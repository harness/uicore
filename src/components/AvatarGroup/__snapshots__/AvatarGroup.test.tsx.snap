// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render basic component should check snapshot with Avatar props with add button 1`] = `
<div>
  <div
    class="stack"
    style="--avatar-over-lap: -13px;"
  >
    <div
      class="stackAvatar Avatar contentStyle"
    >
      <span
        class="bp3-popover-wrapper avatarPopOver"
      >
        <span
          class="bp3-popover-target"
        >
          <div
            class="AvatarInner"
            style="border-radius: 100%; line-height: 26px; width: 32px; height: 32px; font-size: 10px;"
            tabindex="0"
          >
            +
          </div>
        </span>
      </span>
    </div>
    <div
      class="stackAvatar Avatar contentStyle"
    >
      <span
        class="bp3-popover-wrapper avatarPopOver"
      >
        <span
          class="bp3-popover-target"
        >
          <div
            class="AvatarInner"
            style="border-radius: 100%; line-height: 26px; width: 32px; height: 32px; font-size: 10px;"
            tabindex="0"
          >
            <img
              alt=""
              class="imageStyle"
              src="https://image.shutterstock.com/image-photo/buddha-statue-hyderabadhyderabadindia-260nw-1366155431.jpg"
              style="border-radius: 100%; line-height: 26px; width: 28px; height: 28px; font-size: 10px;"
            />
          </div>
        </span>
      </span>
    </div>
    <div
      class="stackAvatar Avatar contentStyle"
    >
      <span
        class="bp3-popover-wrapper avatarPopOver"
      >
        <span
          class="bp3-popover-target"
        >
          <div
            class="AvatarInner"
            style="border-radius: 100%; line-height: 26px; width: 32px; height: 32px; font-size: 10px;"
            tabindex="0"
          >
            GH
          </div>
        </span>
      </span>
    </div>
    <div
      class="stackAvatar Avatar contentStyle"
    >
      <span
        class="bp3-popover-wrapper avatarPopOver"
      >
        <span
          class="bp3-popover-target"
        >
          <div
            class="AvatarInner"
            style="border-radius: 100%; line-height: 26px; width: 32px; height: 32px; font-size: 10px;"
            tabindex="0"
          >
            EF
          </div>
        </span>
      </span>
    </div>
    <div
      class="stackAvatar Avatar contentStyle"
    >
      <span
        class="bp3-popover-wrapper avatarPopOver"
      >
        <span
          class="bp3-popover-target"
        >
          <div
            class="AvatarInner"
            style="border-radius: 100%; line-height: 26px; width: 32px; height: 32px; font-size: 10px;"
            tabindex="0"
          >
            CH
          </div>
        </span>
      </span>
    </div>
    <div
      class="stackAvatar Avatar contentStyle"
    >
      <span
        class="bp3-popover-wrapper avatarPopOver"
      >
        <span
          class="bp3-popover-target"
        >
          <div
            class="AvatarInner"
            style="border-radius: 100%; line-height: 26px; width: 32px; height: 32px; font-size: 10px;"
            tabindex="0"
          >
            AS
          </div>
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`Render basic component should check snapshot with Avatar props without add button 1`] = `
<div>
  <div
    class="stack"
    style="--avatar-over-lap: -13px;"
  >
    <div
      class="stackAvatar Avatar contentStyle"
    >
      <span
        class="bp3-popover-wrapper avatarPopOver"
      >
        <span
          class="bp3-popover-target"
        >
          <div
            class="AvatarInner"
            style="border-radius: 100%; line-height: 26px; width: 32px; height: 32px; font-size: 10px;"
            tabindex="0"
          >
            <img
              alt=""
              class="imageStyle"
              src="https://image.shutterstock.com/image-photo/buddha-statue-hyderabadhyderabadindia-260nw-1366155431.jpg"
              style="border-radius: 100%; line-height: 26px; width: 28px; height: 28px; font-size: 10px;"
            />
          </div>
        </span>
      </span>
    </div>
    <div
      class="stackAvatar Avatar contentStyle"
    >
      <span
        class="bp3-popover-wrapper avatarPopOver"
      >
        <span
          class="bp3-popover-target"
        >
          <div
            class="AvatarInner"
            style="border-radius: 100%; line-height: 26px; width: 32px; height: 32px; font-size: 10px;"
            tabindex="0"
          >
            GH
          </div>
        </span>
      </span>
    </div>
    <div
      class="stackAvatar Avatar contentStyle"
    >
      <span
        class="bp3-popover-wrapper avatarPopOver"
      >
        <span
          class="bp3-popover-target"
        >
          <div
            class="AvatarInner"
            style="border-radius: 100%; line-height: 26px; width: 32px; height: 32px; font-size: 10px;"
            tabindex="0"
          >
            E
          </div>
        </span>
      </span>
    </div>
    <div
      class="stackAvatar Avatar contentStyle"
    >
      <span
        class="bp3-popover-wrapper avatarPopOver"
      >
        <span
          class="bp3-popover-target"
        >
          <div
            class="AvatarInner"
            style="border-radius: 100%; line-height: 26px; width: 32px; height: 32px; font-size: 10px;"
            tabindex="0"
          >
            CH
          </div>
        </span>
      </span>
    </div>
    <div
      class="stackAvatar Avatar contentStyle"
    >
      <span
        class="bp3-popover-wrapper avatarPopOver"
      >
        <span
          class="bp3-popover-target"
        >
          <div
            class="AvatarInner"
            style="border-radius: 100%; line-height: 26px; width: 32px; height: 32px; font-size: 10px;"
            tabindex="0"
          >
            AS
          </div>
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`Render basic component should check snapshot with Avatar props without overlap 1`] = `
<div>
  <div
    class="noStack"
    style="--avatar-over-lap: -13px;"
  >
    <div
      class="Avatar contentStyle"
    >
      <span
        class="bp3-popover-wrapper avatarPopOver"
      >
        <span
          class="bp3-popover-target"
        >
          <div
            class="AvatarInner"
            style="border-radius: 100%; line-height: 26px; width: 32px; height: 32px; font-size: 10px;"
            tabindex="0"
          >
            AS
          </div>
        </span>
      </span>
    </div>
    <div
      class="Avatar contentStyle"
    >
      <span
        class="bp3-popover-wrapper avatarPopOver"
      >
        <span
          class="bp3-popover-target"
        >
          <div
            class="AvatarInner"
            style="border-radius: 100%; line-height: 26px; width: 32px; height: 32px; font-size: 10px;"
            tabindex="0"
          >
            CH
          </div>
        </span>
      </span>
    </div>
    <div
      class="Avatar contentStyle"
    >
      <span
        class="bp3-popover-wrapper avatarPopOver"
      >
        <span
          class="bp3-popover-target"
        >
          <div
            class="AvatarInner"
            style="border-radius: 100%; line-height: 26px; width: 32px; height: 32px; font-size: 10px;"
            tabindex="0"
          >
            E
          </div>
        </span>
      </span>
    </div>
    <div
      class="Avatar contentStyle"
    >
      <span
        class="bp3-popover-wrapper avatarPopOver"
      >
        <span
          class="bp3-popover-target"
        >
          <div
            class="AvatarInner"
            style="border-radius: 100%; line-height: 26px; width: 32px; height: 32px; font-size: 10px;"
            tabindex="0"
          >
            GH
          </div>
        </span>
      </span>
    </div>
    <div
      class="Avatar contentStyle"
    >
      <span
        class="bp3-popover-wrapper avatarPopOver"
      >
        <span
          class="bp3-popover-target"
        >
          <div
            class="AvatarInner"
            style="border-radius: 100%; line-height: 26px; width: 32px; height: 32px; font-size: 10px;"
            tabindex="0"
          >
            <img
              alt=""
              class="imageStyle"
              src="https://image.shutterstock.com/image-photo/buddha-statue-hyderabadhyderabadindia-260nw-1366155431.jpg"
              style="border-radius: 100%; line-height: 26px; width: 28px; height: 28px; font-size: 10px;"
            />
          </div>
        </span>
      </span>
    </div>
  </div>
</div>
`;
